.github/workflows/unit_tests.yml

```diff
@@ -0,0 +1,33 @@
+name: Unit Tests
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  run-tests:
+    name: Run Unit tests
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [windows-latest, macos-latest, ubuntu-latest]
+
+    steps:
+      - name: Check out Git repository
+        uses: actions/checkout@v3
```

Comment:
let's pin to a commit hash for recommended security practices 

actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 (this is v4) as an example 

https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/.github/workflows/custard-ci.yaml#L44C15-L44C72